src/Storages/ObjectStorage/StorageObjectStorageSource.cpp

```diff
@@ -530,7 +531,22 @@ StorageObjectStorageSource::ReaderHolder StorageObjectStorageSource::createReade
                 sample_header.insert({type->createColumn(), type, name});
             }
             initial_header = sample_header;
+            schema_changed = true;
         }
+        auto filter_info = [&]()
+        {
+            if (schema_changed)
+            {
+                return std::make_shared<FormatFilterInfo>(
+                    format_filter_info->filter_actions_dag,
+                    format_filter_info->context.lock(),
+                    configuration->getColumnMapperForObject(object_info));
+            }
+            else
+            {
+                return format_filter_info;
+            }
```

Comment:
```suggestion
            if (schema_changed)
            {
                if (auto mapper = configuration->getColumnMapperForObject(object_info); mapper != nullptr)
                    return std::make_shared<FormatFilterInfo>(
                        format_filter_info->filter_actions_dag,
                        format_filter_info->context.lock(),
                        mapper);
            }
            return format_filter_info;
```

Reply:
> rewrote in a bit different way to take into account that mapper != nullptr

`if (auto mapper = configuration->getColumnMapperForObject(object_info); mapper != nullptr)` does take it into account btw (see second part of the if)